test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -152,21 +154,55 @@ public void CreateBindingMetadata_EditableAttributeTrue_SetsReadOnlyFalse()
             Assert.False(context.BindingMetadata.IsReadOnly);
         }
 
+        // This is IMPORTANT. Product code needs to use GetName() instead of .Name. It's easy to regress.
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_NameFromResources_NullLocalizer()
+        {
+            // Arrange
+            var provider = new DataAnnotationsMetadataProvider(stringLocalizerFactory: null);
+
+            var display = new DisplayAttribute()
+            {
+#if USE_REAL_RESOURCES
+                Name = nameof(Test.Resources.DisplayAttribute_Name),
+                ResourceType = typeof(Test.Resources),
+#else
+                Name = nameof(Test.Resources.DisplayAttribute_Name),
+                ResourceType = typeof(TestResources),
+#endif
```

Comment:
When you say "Use the standard pattern for `Name`" do you mean you want `nameof(DataAnnotations.Test.Resources...)`?